packages/intrepid2/src/Discretization/Basis/Intrepid2_TensorBasis.hpp

```diff
@@ -937,7 +937,7 @@ namespace Intrepid2
               int fieldOrdinal = (fieldOrdinal3 * numFields2_ + fieldOrdinal2) * numFields1_ + fieldOrdinal1;
               for (int pointOrdinal=0; pointOrdinal<numPoints_; pointOrdinal++)
               {
-                output_(fieldOrdinal,pointOrdinal) = weight_ * input1_(fieldOrdinal1,pointOrdinal) * input2_(fieldOrdinal2,pointOrdinal) * input3_(fieldOrdinal3,pointOrdinal);
+                output_(fieldOrdinal,pointOrdinal,0) = weight_ * input1_(fieldOrdinal1,pointOrdinal,0) * input2_(fieldOrdinal2,pointOrdinal,0) * input3_(fieldOrdinal3,pointOrdinal,0);
```

Comment:
The reason why I put the extra arguments comes from an error I got in debug mode. Basically the output of `getValue(..,..,OPERATOR_GRAD)` for a 1d basis has rank 3. This generated an error in some of the lines you pointed out because we were providing two arguments instead of three. In a DynRankView, it is OK to provide more arguments than the rank, but no less.

Anyway, looking at it more carefully, I think I understand better what you where trying to do, however to make it work you have to change the condition
https://github.com/trilinos/Trilinos/blob/master/packages/intrepid2/src/Discretization/Basis/Intrepid2_TensorBasis.hpp#L932
to `if ((input1_.rank() == 2) && (input2_.rank() == 2)) && (input3_.rank()==3))`

Also, there was an error here:
https://github.com/trilinos/Trilinos/blob/master/packages/intrepid2/src/Discretization/Basis/Intrepid2_TensorBasis.hpp#L973
in fact the third argument was provided to `input3_ `instead of `input2_`. That's a bit concerning. I'm wondering if in that case `input2_.extent(2)` is always 1.
Anyway, have a look at the way I fixed it and let me know if you are fine with that.